src/planner/binder/expression/bind_lambda.cpp

```diff
@@ -35,74 +35,63 @@ idx_t GetLambdaParamIndex(const vector<DummyBinding> &lambda_bindings, const Bou
 	return offset;
 }
 
-BindResult ExpressionBinder::BindExpression(LambdaExpression &expr, idx_t depth, const LogicalType &list_child_type,
-                                            optional_ptr<bind_lambda_function_t> bind_lambda_function) {
+void ExtractParameterList(LambdaExpression &expr) {
 
-	// this is for binding JSON
-	if (!bind_lambda_function) {
-		auto lhs_expr = expr.lhs->Copy();
-		OperatorExpression arrow_expr(ExpressionType::ARROW, std::move(lhs_expr), expr.expr->Copy());
-		return BindExpression(arrow_expr, depth);
-	}
+	// extract the lambda parameter list, which is a single column
+	// reference, or a list of column references
 
-	// binding the lambda expression
 	D_ASSERT(expr.lhs);
-	if (expr.lhs->expression_class != ExpressionClass::FUNCTION &&
-	    expr.lhs->expression_class != ExpressionClass::COLUMN_REF) {
-		throw BinderException(
-		    "Invalid parameter list! Parameters must be comma-separated column names, e.g. x or (x, y).");
-	}
+	bool valid_param_list = true;
 
-	// move the lambda parameters to the params vector
 	if (expr.lhs->expression_class == ExpressionClass::COLUMN_REF) {
+		// single column reference
 		expr.params.push_back(std::move(expr.lhs));
```

Comment:
Can we turn `expr.params` into a `vector<string>`, or is there a reason we are keeping the column references around?

In general I think it is not great to store this bind information in the `LambdaExpression` directly, though, as it's somewhat confusing in the code to have two versions of the `LambdaExpression` class; one where `ExtractParameterList` has been called and one where it has not. Perhaps this can be taken out of the `LambdaExpression` class entirely?